### ISSUE TITLE
saw-like segmentation edges & bad performance on real images

### ISSUE THREAD

**Issue by mszuyx** [zhukaii/OS-TR#1]
Hi! 

First, we are really impressed by your one-shot texture retrieval work! We would like to congratulate you and your team on the amazing results!

We are a research team in UIUC and we are in a search for good texture segmentation solutions. I think OS-TR has great potential.

**Status:**

Right now, we are able to train the model using _train.py_ provided in this repo. Using DTD dataset, and this setting (batch_size = 8, epoch = 200, default SGD & learning rate scheduler), we are able to achieve a mean IoU of ~0.521 during training.

_train - IoU_mean   0.498
train - IoU_mean_detail [0.71984734 0.79095859 0.44423657 0.44516743 0.44234881 0.36764969]
train - FBIoU      0.603
train - best_epoch:111
train - best_IoU   0.521_

And here are some validation results from DTD:
![Figure_01](https://user-images.githubusercontent.com/37651144/143141627-32cbe9b7-68ad-4fac-86f8-6cb4ca515ed3.png)
![Figure_11](https://user-images.githubusercontent.com/37651144/143141628-0ad45b97-a7d7-4df7-b475-677a6d9962fe.png)

**Questions:**

1. The model seems to work / respond with synthesized examples from DTD after merely 200 epochs of training. However, one thing that caught our attention is the saw-like coarse edges on the segmentation boundary (as if the segmentation resolution is really low, which is not the case with the original paper). The squares seem to suggest all the pixels within a certain square share the same likelihood of being the positive / negative prediction. As we only started learning your model for a few weeks and didn't change anything special in your scripts, it will be really helpful if we can get your insights on this problem!
2. We also tested the trained model with some real images. The performance is quite different from our expectation since this is a relatively easy test case. Should we be expecting this? -- Since the paper suggests a 1000 epochs training course and it seems like some of the default hyper parameters are also different from the paper's recommendation. What is a "good" number on the _best_IoU_ for the trained model? Could you share your parameters on the SGD and learning rate scheduler?

Thanks! :  )
![Figure_0](https://user-images.githubusercontent.com/37651144/143142313-014d7dc1-18b3-4af8-99c5-41a02abb343a.png)
![Figure_1](https://user-images.githubusercontent.com/37651144/143142316-a92f2b6f-ed81-4fc5-bf27-a844123e15a2.png)

**Comment by mszuyx** [comment]
By specifying the interpolation mode in _'mymodel_dtd.py'_ (line 498 - 512) seems to fix this issue:

```
        img = F.interpolate(img, 16, mode='bilinear', align_corners=False)

        img = torch.cat([img, img3], dim=1)
        img = self.decode2(img)
        img = F.interpolate(img, 32, mode='bilinear', align_corners=False)

        img = torch.cat([img, img2], dim=1)
        img = self.decode3(img)
        img = F.interpolate(img, 64, mode='bilinear', align_corners=False)

        img = torch.cat([img, img1], dim=1)
        img = self.decode4(img)
        img = F.interpolate(img, 256, mode='bilinear', align_corners=False)

        img = torch.sigmoid(img)
```
![Figure_02](https://user-images.githubusercontent.com/37651144/143640930-f7f9ff62-dd57-4dbd-94c9-82900522e164.png)
